pkg/typesystem/typesystem.go

```diff
@@ -557,7 +571,7 @@ func isUsersetRewriteValid(
 
 func validateRelationTypeRestrictions(model *openfgapb.AuthorizationModel) error {
```

Comment:
I forgot if we talked about this, but should this function just take a typesystem (instead of an auth model)?